### PR TITLE
add security-scan workflow

### PR DESCRIPTION
**Pull request by mcollao-hc**

### CODE REVIEW

File: .github/workflows/security-scan.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        echo "$HOME/.bin" >> $GITHUB_PATH
+        echo "$HOME/.bin/codeql" >> $GITHUB_PATH
+
+    - name: Scan
+      id: scan
+      uses: ./security-scanner
+      # env:
+        # Note: this _should_ work, but causes some issues with Semgrep.
+        # Instead, rely on filtering in the SARIF Output step.
+        #SEMGREP_BASELINE_REF: ${{ github.base_ref }}
```
Comment (johanbrandhorst) [inline .github/workflows/security-scan.yml:71]: Should we just remove this?

File: scan.hcl [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  osv = true
+  secrets {
+    all = true
+  } 
+  dependabot {
+    required = true
+    check_config = true
+  }
+  
+  plugin "semgrep" {
```
Comment (johanbrandhorst) [inline scan.hcl:12]: I love semgrep, will this allow us to add custom rules as well?

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]

File: .github/workflows/security-scan.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.x
+
+    - name: Clone Security Scanner repo
+      uses: actions/checkout@v3
+      with:
+        repository: hashicorp/security-scanner
+        token: ${{ secrets.HASHIBOT_PRODSEC_GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (mcollao-hc) [reply .github/workflows/security-scan.yml:33]: Just added it.

File: scan.hcl (hunk 2 above)
Comment (mcollao-hc) [reply scan.hcl:12]: Yup, can add any custom ruleset here, https://github.com/hashicorp/boundary/pull/2945/files#diff-5c4fa294b508f98a963471292bd581cab3912953dc491e452488a6a7e1bf4dbaR15.

File: .github/workflows/security-scan.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
+name: Security Scan
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches:
+      - 'main'
+      - '!oss-merge-main*'
```
[the remark this answers is not in the document]
Comment (mcollao-hc) [reply .github/workflows/security-scan.yml:9]: Holdover from Vault. Was trying to get ahead of any oss->ent merges but you're right, not used here.

File: .github/workflows/security-scan.yml (hunk 1 above)
Comment (mcollao-hc) [reply .github/workflows/security-scan.yml]: 👍

**Review by johanbrandhorst — APPROVED** [review]
LGTM, thanks!

**Comment by johanbrandhorst** [comment]
The build failure is a flake but the `make-gen-deltas` error is legit, we need a license header for this new file:

```diff
diff --git a/scan.hcl b/scan.hcl
index 1ce30e1e9..752e6e547 100644
--- a/scan.hcl
+++ b/scan.hcl
@@ -1,3 +1,6 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
 repository {
   go_modules = true
   osv = true
```

File: .github/workflows/security-scan.yml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    runs-on:
+      labels: ['linux', 'large']
+    if: ${{ github.actor != 'dependabot[bot]' || github.actor != 'hc-github-team-secure-boundary' }}
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
```
[the remark this answers is not in the document]
Comment (mcollao-hc) [reply .github/workflows/security-scan.yml]: Added

**Pull request merged**